### PR TITLE
[Login - Backend] Introducing the needed requirements, settings and migrations

### PR DESCRIPTION
**Pull request by fnormand01**
# Description
---
**Note that this PR is based on changes from the following PRs:**
- PR #275 
- PR #276 
- PR #279 
- PR #281 
---
This PR is introducing all the backend requirements, settings and migrations that we'll need to make the new login functionality work.

I added all requirements from the **react-django-login-example** that were missing in our project. I also added and updated our _settings.py_ file according to this project.

Login Project Example Link: https://github.com/opmftw/react-django-login-example

## Type of change

- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor
- Dependency change

## Screenshot

(N/A)

# Testing

Manual steps to reproduce this functionality:

**- Nothing visual has been changed, so just make sure that the app is still working as expected**

# Checklist

Applicable for all code changes.

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new compiler warnings
- [ ] ~~My changes generate no new accessibility errors and/or warnings~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [ ] ~~I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)~~
- [ ] ~~My changes look good on IE 11+ and Chrome~~

### CODE REVIEW

**Comment by fnormand01** [comment]
Just for documentation purposes, here are the versions that were causing vulnerability issues:
- djoser: 1.1.5 --> 1.5.1
- Jinja2: 2.10 --> 2.10.1
- requests: 2.18.4 --> 2.21.0
- urllib3: 1.22 --> 1.25.2

**Comment by fnormand01** [comment]
For documentation purposes, there is currently a warning related to back end dependencies:

image
_(V 1.25)_

This is caused by urllib3 version >=1.25. If we put version < 1.25, this warning disappears, but we get vulnerability issues:

 
image
_(V 1.24.3)_

**More about this vulnerability:** https://snyk.io/vuln/SNYK-PYTHON-URLLIB3-174323

**Comment by fnormand01** [comment]
@caleybrock, to answer your comment, the problem here is that the migration folder was under _backend_ and now it is under _models_, which makes more sense since the models create tables in our DB.

I had to do it that way, because for some reason, my new User model wasn't detected in _backend_, so I had to clear the old DB and remove all migration history.. I mean, restart from nothing. Now that the migrations folder is under _models_, we shouldn't have to do it like that.

So to make it work on the server, I think we'll need to drop all the existing tables and then migrate, but I'm not 100% sure here what will be the behavior.

Hope it answers your questions?

**Comment by fnormand01** [comment]
@caleybrock @MichaelCherry FYI: I'll only merge this PR when other login PRs will be done and approved.

File: frontend/src/tests/modules/LoginRedux.test.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+  });
+  it("should update authenticated state to false", () => {
+    const action = authenticateAction(false);
+    expect(login(initialState, action)).toEqual({
+      authenticated: false
+    });
   });
 });
+
+// TODO(fnormand): Add tests for registerAction, loginAction and logoutAction
```
Comment (caleybrock) [inline frontend/src/tests/modules/LoginRedux.test.js]: Can you make a small PR branching off of this branch with these tests added.

File: backend/config/settings/dev.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from .base import *
+
+
+DEBUG = True
```
[the remark this answers is not in the document]
Comment (fnormand01) [reply backend/config/settings/dev.py]: Fixed in commit #https://github.com/code-for-canada/project-thundercat/pull/261/commits/388a745a72cc6bba17e7ded1851d0456450b4b50

File: frontend/src/tests/modules/LoginRedux.test.js (hunk 1 above)
Comment (fnormand01) [reply frontend/src/tests/modules/LoginRedux.test.js]: I will add these new tests. in a future PR. As discussed offline, this PR is big enough.

**Pull request merged**